### PR TITLE
Refactor project code base 🛠 

### LINKED ISSUE #238: Ranges of refactors and code consistency enforcement in `lib` directory
- [x] Make sure that page/component/* is only used in that page or move it to global component
- [x] Move utils/UI to corresponding components
- [x] Come up with better names for some classes / widgets
- [x] Respect and enforce types and names and local variables in build methods for widgets
- [x] Use correct widget types, and always `HookConsumerWidget` when it is not required
- [x] Overview on every file + project structure

**Note**: PR should be ready by Saturday (05/10) 7PM.

### PR DESCRIPTION
**Pull request by gruvw**
Fixes #238

### Refactor project code base

🎉🚀 Breaking News! **The Great Refactor Strikes Again**! 🚀🎉

In a flurry of commits that rival even the most chaotic spaghetti code, our fearless contributor @gruvw (yes yes myself) has embarked on an epic quest to tidy up our codebase like never before! Behold, the legends of refactor, where directories are shifted, names are corrected, and widgets are standardized to comply with the sacred laws of the ancient lost city **Widgetopia**!

🔧 **Async Helper Hideout**: All async-related helpers now dwell in a single, harmonious directory, where they plot their cunning schemes in unison.

🔑 **Folders Kingdom**: Witness the majesty of renamed directories and properly placed files, where KeyValueListBuilder reigns supreme with its newfound emplacement and pattern!

🔥 **Firebase Flare-Up**: Firebase options have been relocated to their rightful place in the utils directory, no longer lost in the wilderness of our project structure.

🏠 **Home Sweet Home Page**: The home content has evolved into a full-fledged home page, embracing its **locality of behavior** with open arms.

🎨 **Widget Wonderland**: From `PostHeader` to `AuthLoggedInUser`, our widgets have been reborn, following the sacred Widget Build Conventions with unwavering dedication.

🔍 **Validation Voyage**: Join us as we voyage to the validation directory, where data validation classes frolic in the fields of correctness, ensuring our inputs remain pristine.

And as the dust settles on this monumental endeavor, we stand united, basking in the glow of a codebase transformed. But fear not, dear colleagues, the refactor never truly ends! As long as there is code to be written, refactoring missions will keep being assigned, one contributor at a time :) (I'll pass my turn next time xd.)

**P.S.**: try to save @gruvw from the refactor ruins of Widgetopia by approving this PR, so he can finally be rescued and stay clear from this _code hell_ 🔥😈.

#### Refactoring technical specifications

- [X] single directory for async related helper components
- [X] `KeyValueListBuilder` name + emplacement + pattern
- [X] types util to new helper directory
- [X] move user feedback related components to common directory
- [X] `CentauriSnackBar` as a widget class instead of global function
- [X] components/layout directory for layout related helper widgets
- [X] move utils/UI to corresponding components in view
- [X] proxima app to own file in view
- [X] create account page to its own directory
- [X] option widgets grouped in `options` components directory
- [X] home content now with home page (locality of behavior)
- [X] rename home page top bar to `HomeTopBar`
- [X] `CommentCount`, `PostVotes`, `PostHeader` widgets follow widget build + naming conventions
- [X] authentification logged in user to `AuthLoggedInUser`
- [X] standardization of UI related data information classes (`Details` suffix) + file names
- [X] standardization of UI validation related classes (`Validation` suffix) + file names + common directory
- [X] standardization of services related naming (`Service` suffix) + consistent with file names + directory
- [X] fixed documentation comments (from `//` to `///`)
- [X] rename `HomeViewModel` to `PostsFeedViewModel`
- [X] standardization of view model providers related naming (with `ViewModel` and `Provider` suffix)
- [X] use correct widget types, and not always `HookConsumerWidget` when it is not required
- [X] refactor nested private `_CreateAccountPageContent` into a proper `CreateAccountForm` widget
- [X] overview on every file + project structure (minor corrections, moved files and new directories)
- [X] respect and enforce types, names, local variables and build methods for widgets in view
- [X] make sure that page/component/* is only used in that page or move it to global component
- [X] use new directory structure and file names in test

#### Public service announcement

The refactor is certainly not _perfect_ (it was already hard & long enough like that), especially regarding local variable names.  
Future contributors to the code base are welcome to fix missing details as they start building upon those new names and classes.

This is also true, regrading a few classes that are still left undocumented by their original authors.

#### `Pubspec` changes

- The project version was bumped to `v0.3.2` (Sprint 2 of Milestone 3)
- The project now has a proper description: "A social app that encourages users to visit places in order to see posts."

### CODE REVIEW

File: lib/models/ui/user_profile_details.dart [hunk 1]
```diff
@@ -1,13 +1,12 @@
+import "package:proxima/models/auth/auth_logged_in_user.dart";
 import "package:proxima/models/database/user/user_firestore.dart";
-import "package:proxima/models/login_user.dart";
 
-// TODO (model team) convert to freezed
 // TODO see with UI team what will be displayed for profile
```
Comment (camillelnne) [inline lib/models/ui/user_profile_details.dart:4]: Isn't it solved? Everyone in the team seems to have approved of the current UI. 
Else I couldn't find any related task in the backlog, we could create it.

File: lib/services/sorting/post/post_sort_option.dart [hunk 2]
```diff
@@ -0,0 +1,28 @@
+import "package:cloud_firestore/cloud_firestore.dart";
+import "package:proxima/models/database/post/post_firestore.dart";
+import "package:proxima/services/sorting/post/post_score_functions.dart";
+
+typedef PostScoreFunction = double Function(
```
Comment (camillelnne) [inline lib/services/sorting/post/post_sort_option.dart:5]: I think it is a bit misleading that `PostScoreFunction` is defined here and not in `post_score_function.dart`.
More generally speaking, what motivated you to separate `post_score_function.dart` in two files? I thought `hotScore(PostFirestore post, GeoPoint userPosition)` fitted well with the rest.

**Review by camillelnne — COMMENTED** [review]
Thank you very much for your hard work! 
Your refactor really goes into depth and will facilitate so much future development and testing.

Things I found awesome (you detailed it in the description but I cannot not express my amazement while I review):
- Standard class names, eg all classes with names ending with "Details"
- Moving the ProximaApp class out of the main
- Adding documentation and comments
- The structure of `/lib/views` is soooo satisfying especially `/components`
- ... and much more :)

Thanks!!!

File: lib/services/sorting/post/post_sort_option.dart (hunk 2 above)
Comment (gruvw) [reply lib/services/sorting/post/post_sort_option.dart]: Yes I agree with you the typedef of `PostScoreFunction` with the functions: done in 45e2bdf

I split the old `post_sort_option` in two file for three reasons:

1. Stay consistent with "same styles" enums in the project (like `MapSelectionOptions`, `NavigationbarRoutes`, ...)
2. Avoid polluting the global namespace when importing `PostSortOption` with global orphan functions. (instead accessed through the enum)
3. Separation of concerns: those functions could exist and be used without needing the enum so it makes sense to be able to reference them directly (as opposed to making them private for example).

File: lib/models/ui/user_profile_details.dart (hunk 1 above)
Comment (gruvw) [reply lib/models/ui/user_profile_details.dart]: Yes ty, done in 92a8a12

File: lib/services/sorting/post/post_sort_option.dart (hunk 2 above)
Comment (camillelnne) [reply lib/services/sorting/post/post_sort_option.dart]: ok! By curiosity, how could `hotScore` be used outside of the post sorting?

**Review by camillelnne — APPROVED** [review]
Thanks for the changes!
LGTM :)

File: lib/services/sorting/post/post_sort_option.dart (hunk 2 above)
Comment (gruvw) [reply lib/services/sorting/post/post_sort_option.dart]: First of all, you are only talking about `hotScore`, but note that this change was not only for `hotScore`. It was made to regroup all 4 sorting functions together in `post_score_function.dart`: `hotScore`, `voteScore`, `dayScore`, `distanceScore`.

There are many use cases where those could be used without going through the more UI oriented associated enum `PostSortOption`.
To give a few examples, in some kind of data analysis scenario, or if we would like to get the score of posts to filter them, or apply some kind of spam detection, or maybe for the future heatmap, ...

**Pull request merged**